Source/WebCore/platform/PublicSuffixStore.cpp

```diff
@@ -71,30 +72,32 @@ String PublicSuffixStore::publicSuffix(const URL& url) const
     return { };
 }
 
-String PublicSuffixStore::topPrivatelyControlledDomain(const String& host) const
+String PublicSuffixStore::topPrivatelyControlledDomain(StringView host) const
 {
     // FIXME: if host is a URL, we could drop these checks.
     if (host.isEmpty())
         return { };
 
     if (!host.containsOnlyASCII())
-        return host;
+        return host.toString();
```

Comment:
Aren’t we losing an optimization here where we could reuse a passed-in string?

Reply:
True but:
1. More than half of the call sites have a StringView, not a String
2. This is the case where the URL host contains non-ASCII characters, I don't think this is a case we need to optimize for.